src/app/_components/main-wrapper.tsx

```diff
@@ -33,10 +34,11 @@ export default function MainWrapper({ children }: { children: React.ReactNode })
 
     useEffect(() => {
         if (pathname !== currentPath && !isRouting) {
+            setIsTransitioning(true)
```

Comment:
It seems `setIsTransitioning` always has the same value as `isRouting`.

After noticing this, I am just wondering if maybe it would be simpler to just use it instead of creating a new global state?

Reply:
Yes, isPageTransitioning and isRouting essentially track the same thing. I can promote isRouting to be a global state and remove isPageTransitioning if that works better.

Reply:
Ok, it seems a good idea, I mainly wanted to ensure we weren't duplicating some of the states to track the same things.